spec/models/asset_spec.rb

```diff
@@ -330,28 +330,43 @@
       allow(asset).to receive(:etag_from_file).and_return('etag-from-file')
     end
 
-    context "when etag is stored in database" do
-      let(:etag) { 'etag-from-db' }
+    context "when etag attribute is set" do
+      let(:etag) { 'etag-value' }
 
-      it "returns value from database" do
-        expect(asset.etag).to eq('etag-from-db')
+      it "returns etag attribute value" do
+        expect(asset.etag).to eq('etag-value')
       end
     end
 
-    context "when etag is not stored in database" do
+    context "when etag attribute is not explicitly set" do
       let(:etag) { nil }
 
       it "returns value generated from file metadata" do
         expect(asset.etag).to eq('etag-from-file')
       end
+    end
+
+    context "when asset is created" do
+      let(:etag) { nil }
+
+      before do
+        asset.save!
+      end
+
+      it "stores the value generated from the file in the database" do
+        expect(asset.reload[:etag]).to eq('etag-from-file')
```

Comment:
I found it slightly confusing that this is using `#[]` and passing `:etag` instead of calling `#etag` directly (and similarly when calling `#[]` with `:last_modified` and `:md5_hexdigest` elsewhere). Can we call the methods instead?

Reply:
I've addressed this via an interactive rebase in a couple of commits where calls to the `Asset#[]` method were introduced. I plan to force push once I've addressed the other issues.